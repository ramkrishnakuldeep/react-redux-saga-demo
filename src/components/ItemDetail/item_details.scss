@import '../../resources/styles/variables';

.item-details {
    padding-top: 70px;
    margin: 0 auto;
    width: 400px;

    .item-img {
        position: relative;
        img {
            object-fit: contain;
        }
    }

    .details {
        padding: 10px;
        .name {
            font-size: 1.5rem;
            margin-bottom: 20px;
            text-transform: capitalize;
        }
    
        .likes {
            background: $lightgray;
            border-radius: 30px;
            padding: 5px 10px;
            margin-right: 5px;
    
            &>svg {
                padding-right: 5px;
            }
        }
    
        .description {
            margin-top: 20px;
            font-size: 1.5rem;
            text-transform: capitalize;
        }
    }
    
}