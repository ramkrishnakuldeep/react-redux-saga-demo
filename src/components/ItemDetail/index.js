import React from 'react';
import { itemDetails } from '../../propTypes';
import SoldOut from '../Item/SoldOut';
import './item_details.scss';
import ItemHeader from '../ItemHeader';
import ItemFooter from '../ItemFooter';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';


const ItemDetail = ({ itemDetails }) => {
    return (
        <React.Fragment>
            <ItemHeader item={itemDetails} />
            <div className="item-details">
                <div className="item-img">
                    <img src={itemDetails.image} />
                    {
                        itemDetails.is_sold_out && <SoldOut />
                    }
                </div>
                <div className="details">
                    <div className="name"> {itemDetails.name} </div>
                    <div>
                        <span className="likes">
                            <FontAwesomeIcon icon={faHeart} />
                            Liked count
                        </span>
                        {itemDetails.like_count}
                    </div>
                    <div className="description"> {itemDetails.description} </div>
                </div>
            </div>
            <ItemFooter item={itemDetails} />
        </React.Fragment>
    );
};

ItemDetail.propTypes = { itemDetails };

ItemDetail.defaultProps = {
    itemDetails: {}
};

export default ItemDetail;
