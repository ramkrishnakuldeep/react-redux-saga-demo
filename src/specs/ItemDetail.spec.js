import React from 'react';
import { shallow } from '../enzyme';
import ItemDetail from '../components/ItemDetail';
import { MockItem, MockSoldOutItem } from './__mock__/data';

describe('ItemDetail component', () => {
    it('render soldout item', () => {
        const component = shallow(<ItemDetail itemDetails ={MockSoldOutItem} />);
        expect(component).toMatchSnapshot();
    });

    it('render item', () => {
        const component = shallow(<ItemDetail itemDetails ={MockItem} />);
        expect(component).toMatchSnapshot();
    });
});
