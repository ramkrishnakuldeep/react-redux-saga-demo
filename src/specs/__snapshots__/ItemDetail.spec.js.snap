// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail component render item 1`] = `
<Fragment>
  <ItemHeader
    item={
      Object {
        "category_id": 1,
        "comment_count": 59,
        "description": "size free 1",
        "id": "1",
        "image": "http://dummyimage.com/400x400/000/fff?text=men1",
        "is_sold_out": false,
        "like_count": 91,
        "name": "men1",
        "price": 51,
        "shipping_fee": "ΘÇüµûÖΦ╛╝πü┐",
      }
    }
  />
  <div
    className="item-details"
  >
    <div
      className="item-img"
    >
      <img
        src="http://dummyimage.com/400x400/000/fff?text=men1"
      />
    </div>
    <div
      className="details"
    >
      <div
        className="name"
      >
         
        men1
         
      </div>
      <div>
        <span
          className="likes"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f004",
                  "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                ],
                "iconName": "heart",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Liked count
        </span>
        91
      </div>
      <div
        className="description"
      >
         
        size free 1
         
      </div>
    </div>
  </div>
  <ItemFooter
    item={
      Object {
        "category_id": 1,
        "comment_count": 59,
        "description": "size free 1",
        "id": "1",
        "image": "http://dummyimage.com/400x400/000/fff?text=men1",
        "is_sold_out": false,
        "like_count": 91,
        "name": "men1",
        "price": 51,
        "shipping_fee": "ΘÇüµûÖΦ╛╝πü┐",
      }
    }
  />
</Fragment>
`;

exports[`ItemDetail component render soldout item 1`] = `
<Fragment>
  <ItemHeader
    item={
      Object {
        "category_id": 1,
        "comment_count": 59,
        "description": "size free 1",
        "id": "1",
        "image": "http://dummyimage.com/400x400/000/fff?text=men1",
        "is_sold_out": true,
        "like_count": 91,
        "name": "men1",
        "price": 51,
        "shipping_fee": "ΘÇüµûÖΦ╛╝πü┐",
      }
    }
  />
  <div
    className="item-details"
  >
    <div
      className="item-img"
    >
      <img
        src="http://dummyimage.com/400x400/000/fff?text=men1"
      />
      <SoldOut />
    </div>
    <div
      className="details"
    >
      <div
        className="name"
      >
         
        men1
         
      </div>
      <div>
        <span
          className="likes"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f004",
                  "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                ],
                "iconName": "heart",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Liked count
        </span>
        91
      </div>
      <div
        className="description"
      >
         
        size free 1
         
      </div>
    </div>
  </div>
  <ItemFooter
    item={
      Object {
        "category_id": 1,
        "comment_count": 59,
        "description": "size free 1",
        "id": "1",
        "image": "http://dummyimage.com/400x400/000/fff?text=men1",
        "is_sold_out": true,
        "like_count": 91,
        "name": "men1",
        "price": 51,
        "shipping_fee": "ΘÇüµûÖΦ╛╝πü┐",
      }
    }
  />
</Fragment>
`;
